### ISSUE TITLE
"TypeError: qa is not a function" when synchronizing with sql.js in the browser

### ISSUE THREAD

**Issue by comniemeer** [typeorm/typeorm#6131]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[x] `sqljs` (version `1.3.0`)
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.2.24` (I can't go to `latest`, because it currently doesn't work in the browser, see #6110)

**Steps to reproduce or a small repository showing the problem:**

I'm calling `connection.synchronize(true)` to drop the database and re-synchronize it in my app, but when I do this, I get this error for every table I defined in my app:
```
query:  PRAGMA foreign_keys = OFF;
PlatformTools.js:102 query:  BEGIN TRANSACTION
PlatformTools.js:102 query:  SELECT 'DROP VIEW "' || name || '";' as query FROM "sqlite_master" WHERE "type" = 'view'
PlatformTools.js:102 query:  SELECT 'DROP TABLE "' || name || '";' as query FROM "sqlite_master" WHERE "type" = 'table' AND "name" != 'sqlite_sequence'
PlatformTools.js:102 query:  BEGIN TRANSACTION
PlatformTools.js:102 query:  DROP TABLE "my_table";

PlatformTools.js:105 query failed:  DROP TABLE "my_table";
PlatformTools.js:105 error:  TypeError: qa is not a function
    at Mu (sql-asm.js:127)
    at Fu (sql-asm.js:115)
    at Ku (sql-asm.js:127)
    at Array.rs (sql-asm.js:103)
    at tb (sql-asm.js:103)
    at ql (sql-asm.js:103)
    at im (sql-asm.js:103)
    at Va (sql-asm.js:103)
    at e._sqlite3_step (sql-asm.js:216)
    at a.step (sql-asm.js:79)
```

**Comment by SimMue** [comment]
I'm encountering the same problem with:
` 
"sql.js": "^1.3.0",
"typeorm": "^0.2.21",
`

when I downgrade sql.js to sql.js@1.1.0 it works.
Found the solution here #1365 

From my experience, when running into such problems, it's always helpful to play with the version numbers as some versions of TypeORM simply have dependency errors.

**Comment by bolivir** [comment]
It seems to do the job but i got a strange problem. 
When i use the versions you recommended the error disappeared, but the the app is extremely slow on first startup (Chrome only, Safari seems to work fine).

On the latest version of typeorm + sql.js the error appears as soon as i use a oneToOne or manyToMany relation in an entity.
Anyone has a solution for the extremely slow behaviour in chrome?

**Comment by imnotjames** [comment]
This looks like the error is outside of TypeORM.

You can confirm the following by getting a stack using sourcemaps - or by including an example project / repository that exhibits this issue which can be used for debugging.

It seems that `stackRestore` is what `qa` gets set to in the minified `sql-wasm.js`.  Somehow the stack gets restored too many times and that causes the function to become undefined?  That's a guess.

Either way, it does not seem to be an issue caused by TypeORM directly - but given a way of replicating this I'd be happy to confirm.

To note - sql.js runs as expected in our unit testing & as part of node with wasm.

**Comment by bolivir** [comment]
I just downgraded my ionic 5 projects angular version to 9, now it runs fine in chrome with sql.js 1.1.0 and typeorm 0.2.21.
Will see if i can provide a example project with this issue.

**Comment by imnotjames** [comment]
TLDR: I'm very certain that this problem is caused by `sql.js` and the uglification they've done to `sql-asm.js`.  An issue should be opened in their repository about it.

I'm messing with the ionic example for TypeORM to try to bring it a bit more up to date (or figure out if we should just archive it.. leaning towards that..)

When I include the `sql-asm.js` this happens but when I update the webpack config as such:

```
  new webpack.ProvidePlugin({
    'window.SQL': path.join(__dirname, '../node_modules/sql.js/dist/sql-asm-debug.js'),
  })
```

it works perfectly fine - the `-debug` is a non-uglified "debug" version and it's 11MB.  In other words, the uglification seems messed up somehow.

Also, for 1.3.0 they suggest you DO NOT use the `-asm` version and instead use the `wasm`.

**Comment by imnotjames** [comment]
SCRATCH THAT - they just released 1.3.2 which fixed this.  I think the build they'd published for 1.3.0 was broken accidentally.

**Comment by imnotjames** [comment]
Closing - works great for me now too. :)

I've also updated the example repo with a PR that should help here.

https://github.com/typeorm/ionic-example/pull/38
